openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/LightGBMUtils.java

```diff
@@ -91,13 +118,30 @@ static CpuArchitecture getCpuArchitecture(final String cpuArchName) {
      */
     static private void loadSharedLibraryFromJar(
             final String sharedLibResourceName,
-            final CpuArchitecture cpuArchitecture
+            final CpuArchitecture cpuArchitecture,
+            final boolean isAlpine
```

Comment:
Instead of giving an extra argument to this function can we substitute cpuArchitecture with a new argument that will be a composition of CpuArchitecture with the distribution (or libc that we are using, I prefer this if possible). The class with this composition could be called Infrastructure. You would put the code that you have here to create the .so path  in that class.

Reply:
I agree, that would make the code more clean. I implemented your suggestion in the latest commit.